src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    transform(copyState, action);
+    result.push({ ...copyState });
+  }
+
+  return result;
+}
+
+function transform(copyState, action) {
+  switch (action.type) {
+    case 'addProperties':
+      Object.assign(copyState, action.extraData);
+
+      break;
+
+    case 'removeProperties':
+      for (const key of action.keysToRemove) {
+        delete copyState[key];
+      }
+
+      break;
+
+    case 'clear':
+      for (const key in copyState) {
+        delete copyState[key];
+      }
```

Comment:
Good job, but need to add defaut case.
[CODE STYLE]: switch/case should always have default case for error handling.